### PR TITLE
[WIP] Set the default work start method to spawn

### LINKED ISSUE #16078: Trainer instantiation leads to hanging Work with fork warnings
### Bug description

Running a work with a Trainer inside hangs

### How to reproduce the bug

```python
import lightning as L

class Finetune(L.LightningWork):
    def run(self):
        L.Trainer()
        print("finished")


app = L.LightningApp(Finetune())
```

### Error messages and logs

```
adrian@quantumgrid ~/r/Finetune-miniLM (main)> lightning run app app.py                                                                                    (mini-lm)
Your Lightning App is starting. This won't take long.
INFO: Your app has started. View it in your browser: http://127.0.0.1:7501/view
/Users/adrian/miniconda3/envs/mini-lm/lib/python3.10/site-packages/lightning/app/core/app.py:490: UserWarning: One of your Flows returned `.ready` as `False`. This feature is not yet enabled so this will be ignored.
  warnings.warn(
objc[11096]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[11096]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.

```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): Work
#- PyTorch Lightning Version (e.g., 1.5.0): x
#- Lightning App Version (e.g., 0.5.2): 1.8.5 and master
#- PyTorch Version (e.g., 1.10): 1.13
#- Python version (e.g., 3.9): 3.10
#- OS (e.g., Linux): MacOS
#- CUDA/cuDNN version: x
#- GPU models and configuration: M1
#- How you installed Lightning(`conda`, `pip`, source): pip install -e .
#- Running environment of LightningApp (e.g. local, cloud): local
```

</details>

### More info

Possibly related to #15699

cc @tchaton @justusschock @awaelchli @borda

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Attempt to fix #16078

This PR tests what happens if we switch to `start_method="spawn"` for creating Work processes.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by hhsecond — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
